rust-tests/src/lib.rs

```diff
@@ -0,0 +1,334 @@
+        let expected_hash = variant_hash(format!("{{\"target_platform\": \"{}\"}}", host_subdir()));
+        let pkg_hash = format!("pkg_hash-1.0.0-{expected_hash}_my_pkg.tar.bz2");
+        let pkg = pkg.display().to_string();
+        assert!(pkg.ends_with(&pkg_hash));
+    }
+
+    #[test]
+    fn test_license_glob() {
+        let tmp = tmp();
+        let rattler_build =
+            rattler().build::<_, _, &str>(recipes().join("globtest"), tmp.as_dir(), None);
+        assert!(rattler_build.is_ok());
+        let pkg = get_extracted_package(tmp.as_dir(), "globtest");
+        assert!(pkg.join("info/licenses/LICENSE").exists());
+        assert!(pkg.join("info/licenses/cmake/FindTBB.cmake").exists());
+        assert!(pkg.join("info/licenses/docs/ghp_environment.yml").exists());
+        assert!(pkg.join("info/licenses/docs/rtd_environment.yml").exists());
+        // check total count of files
+        // 4 + 2 folder = 6
+        let path = std::env::current_dir().unwrap();
+        _ = std::env::set_current_dir(pkg);
+        let glen = glob::glob("info/licenses/**/*")
+            .unwrap()
+            .filter(|s| s.is_ok())
+            .count();
+        _ = std::env::set_current_dir(path);
+        assert_eq!(glen, 6);
+    }
+
+    fn check_info(folder: PathBuf, expected: PathBuf) {
+        for f in ["index.json", "about.json", "link.json", "paths.json"] {
+            let expected = expected.join(f);
+            // println!("expected = {}", expected.display());
+            let mut cmp: HashMap<String, serde_json::Value> =
+                serde_json::from_slice(&std::fs::read(expected).unwrap()).unwrap();
+
+            let actual_path = folder.join("info").join(f);
+            assert!(actual_path.exists());
+            // println!("actual = {}", actual_path.display());
+            let actual: HashMap<String, serde_json::Value> =
+                serde_json::from_slice(&std::fs::read(actual_path).unwrap()).unwrap();
+
+            if f == "index.json" {
+                cmp.insert("timestamp".to_string(), actual["timestamp"].clone());
+            }
+            if f == "paths.json" {
+                let act_arr = actual["paths"].as_array().unwrap();
+                let cmp_arr = cmp["paths"].as_array().unwrap();
+                assert!(act_arr.len() == cmp_arr.len());
+                for (i, p) in act_arr.iter().enumerate() {
+                    let c = cmp_arr[i].as_object().unwrap();
+                    let p = p.as_object().unwrap();
+                    assert!(c["_path"] == p["_path"]);
+                    assert!(c["path_type"] == p["path_type"]);
+                    if p["_path"]
+                        .as_str()
+                        .map(|s| s.contains("dist-info"))
+                        .unwrap_or_default()
+                    {
+                        // TODO: figure out why this is not the same b/w expected and cmp
+                        // assert!(c["sha256"] == p["sha256"]);
```

Comment:
everything else is working except, can't figure out why the sha256/hash is different here. hmmm...

Reply:
I can have a look!

Reply:
I think the logic was "backwards" - the distinfo files are the ones that embed the build path and are therefore not deterministic.